### ISSUE TITLE
Zigbee?

### ISSUE THREAD

**Issue by noelhibbard** [koush/scrypted#5]
I'm looking at the plugins folder and don't see any references to Zigbee. Is this something you may add? I have a ton of Zigbee devices. I'm currently using Zigbee2MQTT, does Scrypted have an easy way to integrate via MQTT?

**Comment by noelhibbard** [comment]
Okay, I just noticed the link to the plugin docs. Looks like it wouldn't be too hard to write a plugin to add support for Zigbee2MQTT.

**Comment by koush** [comment]
I’ve been looking at mqtt, but haven’t gotten around to doing it yet since don’t have any mqtt devices. I get the general gist of the pubsub mechanism, but is there a data standard on the endpoint names for say, a lights, and what payload is published and posted? different client/publishers seemed to vary and required different payload handlers, from what I saw.

**Comment by koush** [comment]
I do have a zwave stick that i have a plug-in for, that I could put behind zwave2mqtt…

**Comment by koush** [comment]
Stop in on the discord and I can help you get started with vscode. The docs should be generally up to date though.

**Comment by noelhibbard** [comment]
MQTT itself has no structure at all. You can put any string you want in the payload. A lot of projects that make use of MQTT use JSON for the payload. Zigbee2MQTT has some really good documentation on how the MQTT payloads need to look.

As for your Zwave stick. Zwave and Zigbee are not the same thing. There is a list of supported adapters here, some are as cheap as $12. There are tons of cheap Zigbee devices out there too. For example all the Ikea smart stuff uses Zigbee.:
https://www.zigbee2mqtt.io/information/supported_adapters.html

I've just found Zigbee devices to work more consistently and seem to have less latency when sending commands. I have a few Zwave devices and a lot of times I send a command to them and they don't respond at all or they take 3 secs or so to respond. My Zigbee stuff responds instantly.

**Comment by koush** [comment]
Yes, I know zwave and signee are not the same thing. I was just saying I could use the zwave2mqtt project to build/test mqtt, since I have around 70 zwave devices here.

**Comment by koush** [comment]
> A lot of projects that make use of MQTT use JSON for the payload. Zigbee2MQTT has some really good documentation on how the MQTT payloads need to look.

so a generalized mqtt plugin would likely need to allow the user to specify the publishing topics and parse the meaning of the payload?

**Comment by noelhibbard** [comment]
> > A lot of projects that make use of MQTT use JSON for the payload. Zigbee2MQTT has some really good documentation on how the MQTT payloads need to look.
> 
> so a generalized mqtt plugin would likely need to allow the user to specify the publishing topics and parse the meaning of the payload?



> Yes, I know zwave and zigbee are not the same thing. I was just saying I could use the zwave2mqtt project to build/test an mqtt plugin, since I have around 70 zwave devices here. None of my other hardware talks mqtt.

Oh sorry.. I misread zwave2MQTT as Zigbee2MQTT. I've not looked at the zwave2mqtt project. I suspect the payloads wouldn't be close enough to have a generic plugin that cover both platforms. Although I do think a generic MQTT plugin would be better than one specific to Zigbee2MQTT and another one specific to Zwave2MQTT.

There is a nice MQTT plugin for homebridge called homebridge-mqttthing that you could get ideas from. I use it for a lot of projects of my own. For example I have an RPi in the garage that runs my garage door and sprinklers. To bridge these things into HomeKit I simply added a MQTT interface and then used the homebridge-mqttthing plugin to talk top them. So many projects are using MQTT these days. Frigate NVR is another project that comes to mind that you could tie into with MQTT.

**Comment by koush** [comment]
Have started MQTT support. Have included the Aedes broker.
Right now it provides an easy way to write scripts that manage MQTT subscriptions. Or create domain specific languages to create common devices:

<img width="1119" alt="Screen Shot 2021-09-23 at 11 12 41 PM" src="https://user-images.githubusercontent.com/73924/134627085-6ced46fd-5eac-46af-bb00-e8696b0bed36.png">

Need to eventually add support for HASS autodiscovery protocol.

**Comment by koush** [comment]
@noelhibbard have an initial implementation of mqtt auto discovery. what sort of devices do you have a on zigbee?

**Comment by DedMsk** [comment]
WOW! It's great!
Maybe I can add doorbell sensor from MQTT? )
Can you help me?
